src/probability_theory/independence.lean

```diff
@@ -0,0 +1,312 @@
+  indep_sets s₁ s₃ μ :=
+λ t1 t2 ht1 ht2, h_indep t1 t2 ht1 (set.mem_of_subset_of_mem h32 ht2)
+
+lemma indep_of_indep_of_le_left {α} {m₁ m₂ m₃: measurable_space α} [measurable_space α]
+  {μ : measure α} (h_indep : indep m₁ m₂ μ) (h31 : m₃ ≤ m₁) :
+  indep m₃ m₂ μ :=
+λ t1 t2 ht1 ht2, h_indep t1 t2 (h31 _ ht1) ht2
+
+lemma indep_of_indep_of_le_right {α} {m₁ m₂ m₃: measurable_space α} [measurable_space α]
+  {μ : measure α} (h_indep : indep m₁ m₂ μ) (h32 : m₃ ≤ m₂) :
+  indep m₁ m₃ μ :=
+λ t1 t2 ht1 ht2, h_indep t1 t2 ht1 (h32 _ ht2)
+
+lemma indep_sets.union {α} [measurable_space α] {s₁ s₂ s' : set (set α)} {μ : measure α}
+  (h₁ : indep_sets s₁ s' μ) (h₂ : indep_sets s₂ s' μ) :
+  indep_sets (s₁ ∪ s₂) s' μ :=
+begin
+  intros t1 t2 ht1 ht2,
+  cases (set.mem_union _ _ _).mp ht1 with ht1₁ ht1₂,
+  { exact h₁ t1 t2 ht1₁ ht2, },
+  { exact h₂ t1 t2 ht1₂ ht2, },
+end
+
+@[simp] lemma indep_sets.union_iff {α} [measurable_space α] {s₁ s₂ s' : set (set α)}
+  {μ : measure α} :
+  indep_sets (s₁ ∪ s₂) s' μ ↔ indep_sets s₁ s' μ ∧ indep_sets s₂ s' μ :=
+⟨λ h, ⟨indep_sets_of_indep_sets_of_le_left h (set.subset_union_left s₁ s₂),
+    indep_sets_of_indep_sets_of_le_left h (set.subset_union_right s₁ s₂)⟩,
+  λ h, indep_sets.union h.left h.right⟩
+
+lemma indep_sets.Union {α ι} [measurable_space α] {s : ι → set (set α)} {s' : set (set α)}
+  {μ : measure α} (hyp : ∀ n, indep_sets (s n) s' μ) :
+  indep_sets (⋃ n, s n) s' μ :=
+begin
+  intros t1 t2 ht1 ht2,
+  rw set.mem_Union at ht1,
+  cases ht1 with n ht1,
+  exact hyp n t1 t2 ht1 ht2,
+end
+
+lemma indep_sets.inter {α} [measurable_space α] {s₁ s' : set (set α)} (s₂ : set (set α))
+  {μ : measure α} (h₁ : indep_sets s₁ s' μ) :
+  indep_sets (s₁ ∩ s₂) s' μ :=
+λ t1 t2 ht1 ht2, h₁ t1 t2 ((set.mem_inter_iff _ _ _).mp ht1).left ht2
+
+lemma indep_sets.Inter {α ι} [measurable_space α] {s : ι → set (set α)} {s' : set (set α)}
+  {μ : measure α} (h : ∃ n, indep_sets (s n) s' μ) :
+  indep_sets (⋂ n, s n) s' μ :=
+by {intros t1 t2 ht1 ht2, cases h with n h, exact h t1 t2 (set.mem_Inter.mp ht1 n) ht2 }
+
+end indep
+
+/-! ### Deducing `indep` from `Indep` -/
+section from_Indep_to_indep
+
+lemma Indep_sets.indep_sets {α ι} {s : ι → set (set α)} [measurable_space α] {μ : measure α}
+  (h_indep : Indep_sets s μ) {i j : ι} (hij : i ≠ j) :
+  indep_sets (s i) (s j) μ :=
+begin
+  intros t₁ t₂ ht₁ ht₂,
+  have hf_m : ∀ (x : ι), x ∈ {i, j} → (ite (x=i) t₁ t₂) ∈ s x,
+  { intros x hx,
+    cases finset.mem_insert.mp hx with hx hx,
+    { simp [hx, ht₁], },
+    { simp [finset.mem_singleton.mp hx, hij.symm, ht₂], }, },
+  have h1 : t₁ = ite (i = i) t₁ t₂, by simp only [if_true, eq_self_iff_true],
+  have h2 : t₂ = ite (j = i) t₁ t₂, by simp only [hij.symm, if_false],
+  have h_inter : (⋂ (t : ι) (H : t ∈ ({i, j} : finset ι)), ite (t = i) t₁ t₂)
+      = (ite (i = i) t₁ t₂) ∩ (ite (j = i) t₁ t₂),
+    by simp only [finset.bInter_singleton, finset.bInter_insert],
```

Comment:
The build is failing because #5813 renamed some lemmas. 

Try merging master and then test out this suggestion:
```suggestion
    by simp only [finset.set_bInter_singleton, finset.set_bInter_insert],
```
Then feel free to merge again whenever things are fixed.
bors r-